lightning/src/onion_message/messenger.rs

```diff
@@ -1043,20 +1081,41 @@ where
 		)
 	}
 
-	fn handle_onion_message_response<T: OnionMessageContents>(
+	/// Handles the response to an [`OnionMessage`] based on its [`ResponseInstruction`],
+	/// enqueueing any response for sending.
+	pub fn handle_onion_message_response<T: OnionMessageContents>(
```

Comment:
Let's add further information on how to use this in a second paragraph. Something like:

```rust
/// Useful for handling [`OnionMessage`]s asynchronously. A handler may return
/// [`ResponseInstruction::NoResponse`] and move the passed [`Responder`] to another task for
/// generating a response. That task can then call this method once ready to send the response.
```

Reply:
Sounds good!
Expanded the docs in [834779d](https://github.com/lightningdevkit/rust-lightning/commit/834779de16284f0da3c47b42ca9ac6cd004a142e)
Let me know if it looks good!
Thanks!